snuba/web/rpc/v1/endpoint_get_traces.py

```diff
@@ -0,0 +1,405 @@
+
+class EndpointGetTraces(RPCEndpoint[GetTracesRequest, GetTracesResponse]):
+    @classmethod
+    def version(cls) -> str:
+        return "v1"
+
+    @classmethod
+    def request_class(cls) -> Type[GetTracesRequest]:
+        return GetTracesRequest
+
+    @classmethod
+    def response_class(cls) -> Type[GetTracesResponse]:
+        return GetTracesResponse
+
+    def _execute(self, in_msg: GetTracesRequest) -> GetTracesResponse:
+        _validate_order_by(in_msg)
+
+        in_msg.meta.request_id = getattr(in_msg.meta, "request_id", None) or str(
+            uuid.uuid4()
+        )
+        response_meta = extract_response_meta(
+            in_msg.meta.request_id,
+            in_msg.meta.debug,
+            [],
+            [self._timer],
+        )
+
+        # Get a dict of trace IDs and timestamps.
+        trace_ids = self._list_trace_ids(request=in_msg)
+        if len(trace_ids) == 0:
+            return GetTracesResponse(meta=response_meta)
+
+        # Get metadata for those traces.
+        traces = self._get_metadata_for_traces(request=in_msg, trace_ids=trace_ids)
+        return GetTracesResponse(
+            traces=traces,
+            page_token=_get_page_token(in_msg, traces),
+            meta=response_meta,
+        )
+
+    def _list_trace_ids(
+        self,
+        request: GetTracesRequest,
+    ) -> dict[str, int]:
+        # Find first span filter.
+        # TODO: support more than one filter.
+        trace_item_filters_expression = trace_item_filters_to_expression(
+            next(
+                f.filter
+                for f in request.filters
+                if f.item_name == TraceItemName.TRACE_ITEM_NAME_EAP_SPANS
+            )
+            if len(request.filters) > 0
+            else TraceItemFilter()
+        )
+        selected_columns: list[SelectedExpression] = [
+            SelectedExpression(
+                name="trace_id",
+                expression=f.cast(
+                    column("trace_id"),
+                    "String",
+                    alias="trace_id",
+                ),
+            ),
+            SelectedExpression(
+                name="_sort_timestamp",
+                expression=f.cast(
+                    column("_sort_timestamp"),
+                    "UInt32",
+                    alias="_sort_timestamp",
+                ),
+            ),
+        ]
+        entity = Entity(
+            key=EntityKey("eap_spans"),
+            schema=get_entity(EntityKey("eap_spans")).get_data_model(),
+            sample=None,
+        )
+        query = Query(
+            from_clause=entity,
+            selected_columns=selected_columns,
+            condition=base_conditions_and(
+                request.meta,
+                trace_item_filters_expression,
+            ),
+            order_by=[
+                OrderBy(
+                    direction=OrderByDirection.DESC,
+                    expression=column("_sort_timestamp"),
+                ),
+            ],
```

Comment:
This completely ignores the order by specified by the request. I'm not sure how you'd implement the rest but I'm fine if you just validate it to only allow sorting by timestamp for now.